ui/browsing/albumpage.go

```diff
@@ -246,7 +248,17 @@ func NewAlbumPageHeader(page *AlbumPage) *AlbumPageHeader {
 				a.page.contr.ShowAlbumInfoDialog(a.albumID, a.titleLabel.String(), a.cover.Image())
 			})
 			info.Icon = theme.InfoIcon()
-			menu := fyne.NewMenu("", queue, playlist, download, info)
+			menuItems := []*fyne.MenuItem{queue, playlist, download, info}
+
+			_, canShare := page.mp.(mediaprovider.SupportsSharing)
+			if canShare {
```

Comment:
It's not obvious, but creating the menu item only if `canShare` is true can lead to a bug where if switching servers from Jellyfin to a Subsonic server, the share option may not appear after connecting to the Subsonic server. Here, and in Tracklist, the menu is only created once and reused if it is shown again. And the pages themselves may also be reused when navigating between pages. (See the WidgetPool component in ui/util). This is because creating widgets anew in Fyne is a somewhat expensive operation, and reuse leads to memory and performance improvements.

All this is to say, what we should do instead is always create the sharing menu item, but disable it if `!canShare`. The disabling check should be done every time before it is shown (ie just after the `if pop == nil` block that creates the menu for the first time).

Reply:
This was very tricky, but I think I covered all scenarios where `pop == nil` or `menu == nil` could cache the sharing status!